### DISCUSSION TITLE
Question about sld format

### DISCUSSION THREAD

**Discussion by yaute74** [SFTtech/openage#1714] [General]
Hello,
I have a question about sld format. I want to create a mod for AOE2 DE and I saw that you gave a description of the structure of file.
What I want to do:
- extract a sld file
- modify manually the images
- recreate the sld file

Do you have a tools in openage to do that ?

Thanks

**Comment by heinezen** [comment]
Hey, 

openage only supports extracting the SLD file because we don't use the format in our engine. Therefore, only step 1 and 2 would be possible. Our idea was that the (re)creation of SLD files was better handled by existing tools like [SLX Studio](https://ageofempires.fandom.com/wiki/SLX_Studio). However, I currently don't know the implementation status of the feature, so I'm not sure if it's possible to do that yet.

You can theoretically implement the recreation of the file yourself if you really want, but it's a bit complicated due to the compression algorithms that they used in SLD (i.e. BC1 and BC4). I'm not sure if somebody has written a free library to create compressed images with these formats, although I assume that there are options, given that they are popular formats. So theoretically, you can do this yourself depending on your programming knowledge :)

**Comment by yaute74** [comment]
Actually SLX Studio does not support sld.
Thanks for the information.
Have a good day
